metadata-editor/app/lib/Syndication.scala

```diff
@@ -1,67 +1,82 @@
 package lib
 
 import com.gu.mediaservice.lib.aws.{DynamoDB, NoItemFound, UpdateMessage}
+import com.gu.mediaservice.lib.logging.GridLogging
 import com.gu.mediaservice.model.{Edits, Photoshoot, SyndicationRights}
 import com.gu.mediaservice.syntax.MessageSubjects
-import play.api.libs.json.{JsNull, JsString}
+import org.joda.time.DateTime
+import play.api.libs.json.{JsNull, JsString, Reads}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-trait Syndication extends Edit with MessageSubjects {
+trait Syndication extends Edit with MessageSubjects with GridLogging {
 
   def syndicationStore: SyndicationStore
 
   private val syndicationRightsFieldName = "syndicationRights"
 
-  def deletePhotoshootAndPublish(id: String)
-                                (implicit ec: ExecutionContext): Future[Unit.type] = {
+  private def timedFuture[T](label: String, f : => Future[T])(implicit ec: ExecutionContext): Future[T] = {
+    val timeBefore = DateTime.now().getMillis
+    val result = f
+      result.onComplete { _ =>
+        val timeAfter = DateTime.now().getMillis
+        logger.info(s"$label took ${timeAfter - timeBefore} milliseconds")
+      }
+    result
+  }
+
+  private[lib] def publishChangedSyndicationRightsMessages[T](id: String, unchangedPhotoshoot: Boolean = false, photoshoot: Option[Photoshoot] = None)(f: () => Future[T])
```

Comment:
minor: Could the following renames make this easier to parse?:
  - `f` as `updatePhotoshoot` (??)
  - `before` as `current`
  - `after` as `updated`
 
That is, it would become more of a narrative:
  1. Get current state
  2. Perform update
  3. Get updated state